src/libpgmoneta/message.c

```diff
@@ -552,11 +562,518 @@ pgmoneta_create_startup_message(char* username, char* database, struct message**
+   else
+   {
+      if (xlogpos != NULL && strlen(xlogpos) > 0)
+      {
+         snprintf(&cmd[0], sizeof(cmd), "START_REPLICATION PHYSICAL %s;", xlogpos);
+      }
+      else
+      {
+         snprintf(&cmd[0], sizeof(cmd), "START_REPLICATION PHYSICAL 0/0 TIMELINE %d;", timeline);
+      }
+   }
+
+   size = 1 + 4 + strlen(cmd) + 1;
+
+   m = (struct message*)malloc(sizeof(struct message));
+   m->data = malloc(size);
+
+   memset(m->data, 0, size);
+
+   m->kind = 'Q';
+   m->length = size;
+
+   pgmoneta_write_byte(m->data, 'Q');
+   pgmoneta_write_int32(m->data + 1, size - 1);
+   memcpy(m->data + 5, &cmd[0], strlen(cmd));
+
+   *msg = m;
+
+   return MESSAGE_STATUS_OK;
+}
+
+int
+pgmoneta_create_standby_status_update_message(int64_t received, int64_t flushed, int64_t applied, struct message** msg)
+{
+   struct message* m = NULL;
+   size_t size;
+
+   size = 1 + 8 + 8 + 8 + 8 + 1;
+
+   m = (struct message*)malloc(sizeof(struct message));
+   m->data = malloc(size);
+
+   memset(m->data, 0, size);
+
+   m->kind = 'r';
+   m->length = size;
+
+   pgmoneta_write_byte(m->data, 'r');
+   pgmoneta_write_int64(m->data + 1, received);
+   pgmoneta_write_int64(m->data + 9, flushed);
+   pgmoneta_write_int64(m->data + 17, applied);
+   pgmoneta_write_int64(m->data + 25, pgmoneta_get_current_timestamp() - pgmoneta_get_y2000_timestamp());
+   pgmoneta_write_byte(m->data + 33, 0);
+
    *msg = m;
 
    return MESSAGE_STATUS_OK;
 }
 
+/* void* */
+/* pgmoneta_get_column(int col, struct tuple* tuple) */
+/* { */
+/*    return tuple->data[col]; */
+/* } */
+
+/* void* */
+/* pgmoneta_get_column_by_name(char* name, struct query* query, struct tuple* tuple) */
+/* { */
+/*    for (int i = 0; i < query->number_of_columns; i++) */
+/*    { */
+/*       if (!strcmp(query->names[i], name)) */
+/*       { */
+/*          return pgmoneta_get_column(i, tuple); */
+/*       } */
+/*    } */
+
+/*    return NULL; */
+/* } */
+
+int
+pgmoneta_query_execute(int socket, struct message* msg, struct query** query)
+{
+   int status;
+   int fd = -1;
+   bool cont;
+   int cols;
+   char* name = NULL;
+   struct message* tmsg = NULL;
+   char* content = NULL;
+   struct message* reply = NULL;
+   struct query* q = NULL;
+   struct tuple* current = NULL;
+   size_t data_size;
+   void* data = pgmoneta_memory_dynamic_create(&data_size);
+   size_t offset = 0;
+
+   *query = NULL;
+
+   status = pgmoneta_write_message(NULL, socket, msg);
+   if (status != MESSAGE_STATUS_OK)
+   {
+      goto error;
+   }
+
+   cont = true;
+   while (cont)
+   {
+      status = pgmoneta_read_block_message(NULL, socket, &reply);
+
+      if (status == MESSAGE_STATUS_OK)
+      {
+         data = pgmoneta_memory_dynamic_append(data, data_size, reply->data, reply->length, &data_size);
+
+         if (pgmoneta_has_message('Z', data, data_size))
+         {
+            cont = false;
+         }
+      }
+      else if (status == MESSAGE_STATUS_ZERO)
+      {
+         SLEEP(1000000L);
```

Comment:
Just a plain question, what does `MESSAGE_STATUS_ZERO` mean and why do we sleep here?